Kubernetes 1.5.2 rebase

soltysh opened the pull request:
@ncdc @deads2k @liggitt @smarterclayton ptal
@derekwaynecarr ptal the reverted UPSTREAM prs you asked for

The k8s branch is here: https://github.com/soltysh/kubernetes/tree/openshift-1.5

soltysh commented:
[test]

deads2k commented:
wow, that went amazingly well.

soltysh commented:
> wow, that went amazingly well.

aside from the fact that I can't `go get go4.org/errorutil` today, but it worked yesterday somehow - yes 😃

ncdc commented:
This isn't the first time go4.org has had issues

soltysh commented:
Updated commit messages for the first revert which failed commitchecker.

liggitt commented:
Want to pick https://github.com/kubernetes/kubernetes/issues/39848 or shall I after this merges?

soltysh commented:
> Want to pick kubernetes/kubernetes#39848 or shall I after this merges?

I can add it here.

ncdc commented:
I thought about it more, and I think we do need an openshift-1.5.2 branch in our kube fork, so we don't end up force pushing to the openshift-1.5 branch. Or we need to push a tag for the current HEAD of openshift-1.5, so it's still a valid sha.

liggitt commented:
we've typically tagged and force-pushed the release branch in the kube fork

ncdc commented:
That's fine, as long as there is a tag.

soltysh commented:
This push includes @liggitt request upstream PR and fixes the generated changes. Hopefully this should be good.

ncdc commented:
```
In suite "github.com/openshift/origin/test/cmd/cmd-admin-namespace-setup", test case "hack/test-cmd.sh:257: executing 'oc new-project 'cmd-admin'' expecting success" failed:
=== BEGIN TEST CASE ===
hack/test-cmd.sh:257: executing 'oc new-project 'cmd-admin'' expecting success
FAILURE after 0.471s: hack/test-cmd.sh:257: executing 'oc new-project 'cmd-admin'' expecting success: the command returned the wrong error code
There was no output from the command.
Standard error from the command:
Error from server: no Items field in &unversioned.Status{TypeMeta:unversioned.TypeMeta{Kind:"", APIVersion:""}, ListMeta:unversioned.ListMeta{SelfLink:"", ResourceVersion:""}, Status:"Success", Message:"", Reason:"", Details:(*unversioned.StatusDetails)(nil), Code:0}
```

ncdc commented:
Need to pick https://github.com/kubernetes/kubernetes/pull/39886

soltysh commented:
Rebased, with 39886 picked and updated generated changes.

soltysh commented:
[testextended][extended:all]

soltysh commented:
[testextended][extended:all]

soltysh commented:
Flake #11016 on StatefulSets, need to double check the rest.

[testextended][extended:all]

soltysh commented:
I've rebased on top of master and added the test for `ExperimentalKernelMemcgNotification` being false, as Clayton asked for it. @smarterclayton are you ok with merging this on green tests? I'd really like to not wait too much longer to deliver this 😄

ncdc commented:
+1

On Sun, Jan 15, 2017 at 9:09 AM Maciej Szulik <notifications@github.com>
wrote:

> I've rebased on top of master and added the test for
> ExperimentalKernelMemcgNotification being false, as Clayton asked for it.
> @smarterclayton <https://github.com/smarterclayton> are you ok with
> merging this on green tests? I'd really like to not wait too much longer to
> deliver this 😄
>
>
>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/12481#issuecomment-272697345>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAABYnIptP09gmmLWZYD3YefMsvhznQqks5rSigygaJpZM4Li3IK>
> .
>
>
>
>
>
>
>
>
>
>
>
>
>
>
>
>
>
>
>

soltysh commented:
![](https://github.trello.services/images/mini-trello-icon.png) [(8) Rebase (kube 1.5)](https://trello.com/c/SWNsQvSx/766-8-rebase-kube-1-5)

soltysh commented:
The tests are green, the only failures in extended tests I'm seeing here are also happening in master. I've opened #12499, #12500 and #12501 covering those. Are there any objections in merging this in @smarterclayton @liggitt @deads2k @ncdc ?

ncdc commented:
Merge gets my vote.

On Mon, Jan 16, 2017 at 6:03 AM Maciej Szulik <notifications@github.com>
wrote:

> The tests are green, the only failures in extended tests I'm seeing here
> are also happening in master. I've opened #12499
> <https://github.com/openshift/origin/issues/12499>, #12500
> <https://github.com/openshift/origin/issues/12500> and #12501
> <https://github.com/openshift/origin/issues/12501> covering those. Are
> there any objections in merging this in @smarterclayton
> <https://github.com/smarterclayton> @liggitt <https://github.com/liggitt>
> @deads2k <https://github.com/deads2k> @ncdc <https://github.com/ncdc> ?
>
>
>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/12481#issuecomment-272832293>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAABYvR37AEAGuXvwVtDWxQRIoCR4oB3ks5rS05igaJpZM4Li3IK>
> .
>
>
>
>
>
>
>
>
>
>
>
>
>
>
>
>
>
>
>

deads2k commented on pkg/cmd/server/kubernetes/master.go:
```diff
@@ -118,6 +118,8 @@ func (c *MasterConfig) RunPersistentVolumeAttachDetachController(client *kclient
 			c.CloudProvider,
 			kctrlmgr.ProbeAttachableVolumePlugins(s.VolumeConfiguration),
 			nil,
+			s.DisableAttachDetachReconcilerSync,
```
What are the defaults for these?

soltysh replied:
`ReconcilerSyncLoopPeriod` is 5 sec, `DisableAttachDetachReconcilerSync` is false. The former is tested in pkg/cmd/server/kubernetes/master_config_test.go, I'll add one another for the latter in that case as a followup.

deads2k commented:
[merge]
